clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = railsGirls,
+                                 replaceWith = "Rails Girls")
+
+    ## Normalize "Game Jams" to "Game Jam(s)"
+    gameJams <- c("game.*?jams?")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = gameJams,
+                                 replaceWith = "Game Jam(s)")
+
+
+    # Clean Podcasts Other
+
+    ## Convert Podcasts to binary/boolean
+    podcasts <- cleanPart1 %>%
+        select(starts_with("Podcast"), -PodcastOther) %>%
+        mutate_each(funs(ifelse(!is.na(.), "1", NA)))
+    cleanPart1 <- cleanPart1 %>%
+        select(-starts_with("Podcast"), PodcastOther) %>%
+        bind_cols(podcasts)
+
+    ## Normalize variations of "None" in PodcastOther
+    nonePod <- c("non", "none", "haven't", "havent", "not a", "nothing",
+               "didn't", "n/a", "\bna\b", "never", "nil", "nope", "not tried")
+    nonesPodIdx <- cleanPart1 %>% select(PodcastOther) %>%
+        mutate_each(
+            funs(grepl(paste(nonePod, collapse = "|"),
+                       .,
+                       ignore.case = TRUE)
+            )
+        ) %>% unlist(use.names = FALSE)
+    nonesPodData <- cleanPart1 %>% filter(nonesPodIdx) %>%
+        mutate(PodcastOther = NA) %>%
+        mutate(PodcastNone = "1")
+    cleanPart1 <- cleanPart1 %>% filter(!nonesPodIdx) %>%
+        bind_rows(nonesPodData)
+
+
+    # Clean number of hours spent learning
+
+    ## Remove the word "hour(s)"
+    hoursIdx <- cleanPart1 %>% select(HoursLearning) %>%
+        mutate_each(funs(grepl("hours", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    hoursData <- cleanPart1 %>% filter(hoursIdx) %>%
+        mutate(HoursLearning = sub("hours.*", "", HoursLearning))
+    cleanPart1 <- cleanPart1 %>% filter(!hoursIdx) %>% bind_rows(hoursData)
+
+    ## Remove hyphen and "to" for ranges of hours
+    rangeHrIdx <- cleanPart1 %>% select(HoursLearning) %>%
+        mutate_each(funs(grepl("-|to", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    rangeHrData <- cleanPart1 %>% filter(rangeHrIdx) %>%
+        mutate(HoursLearning = average_string_range(HoursLearning))
+    cleanPart1 <- cleanPart1 %>% filter(!rangeHrIdx) %>%
+        bind_rows(rangeHrData)
+
+    ## Remove hours greater than a week
+    hoursLearnNumeric <- cleanPart1 %>%
+        mutate(HoursLearning = as.numeric(HoursLearning))
+    weekHours <- hoursLearnNumeric %>%
+        filter(HoursLearning > 24*7) %>%
+        mutate(HoursLearning = as.numeric(NA)) # Change to value later
+    cleanPart1 <- hoursLearnNumeric %>%
+        filter(!(ID %in% weekHours[["ID"]])) %>%
+        bind_rows(weekHours) %>%
+        mutate(HoursLearning = as.character(HoursLearning))
+
```

Comment:
@SamAI-Software the diff on the PR doesn't show it but I've limited the hours learning to 100 and made all hours integers :)